src/openforms/forms/tests/test_migrations.py

```diff
@@ -805,3 +805,44 @@ def test_backwards_migration(self):
             self.form3.confirmation_email_option,
             ConfirmationEmailOptions.form_specific_email,
         )
+
+
+class TestReplaceOldCosignComponent(TestMigrations):
+    migrate_from = "0078_form_suspension_allowed"
+    migrate_to = "0079_replace_cosign_component"
+    app = "forms"
+
+    def setUpBeforeMigration(self, apps):
+        FormDefinition = apps.get_model("forms", "FormDefinition")
+        self.form_definition = FormDefinition.objects.create(
+            name="Definition with Co-sign",
+            slug="definition-with-cosign",
+            configuration={
+                "components": [
+                    {
+                        "label": "Co-sign",
+                        "description": "Im a cosign component",
+                        "authPlugin": "digid",
+                        "key": "medeOndertekenen1",
+                        "type": "coSign",
+                    }
+                ]
+            },
+        )
+
+    def test_forward_migration(self):
+        self.form_definition.refresh_from_db()
+
+        cosign_component = self.form_definition.configuration["components"][0]
+
+        self.assertEqual(
+            cosign_component,
+            {
+                "label": "Co-sign",
+                "description": "Im a cosign component",
+                "authPlugin": "digid",
+                "key": "medeOndertekenen1",
+                "type": "cosign",
+                "validateOn": "blur",
+            },
+        )
```

Comment:
If we go with the jsonpath query, we need to test a nested one as well... The `$..` should recurse down into objects and arrays, the way `//` does in XPath.